idea/idea-android/src/org/jetbrains/kotlin/android/quickfix/KotlinAndroidViewConstructorFix.kt

```diff
@@ -0,0 +1,126 @@
+import org.jetbrains.kotlin.resolve.BindingContext
+import org.jetbrains.kotlin.resolve.DescriptorToSourceUtils
+import org.jetbrains.kotlin.resolve.descriptorUtil.fqNameUnsafe
+import org.jetbrains.kotlin.types.KotlinType
+import org.jetbrains.kotlin.types.typeUtil.supertypes
+
+class KotlinAndroidViewConstructorFix(element: KtSuperTypeEntry) : KotlinQuickFixAction<KtSuperTypeEntry>(element) {
+
+    override fun getText() = "Add Android View constructors using '@JvmOverloads'"
+    override fun getFamilyName() = text
+
+    override fun isAvailable(project: Project, editor: Editor?, file: PsiFile): Boolean {
+        if (AndroidFacet.getInstance(file) == null) return false
+        return super.isAvailable(project, editor, file)
+    }
+
+    override fun invoke(project: Project, editor: Editor?, file: KtFile) {
+        val element = element ?: return
+        val ktClass = element.containingClass() ?: return
+
+        val factory = KtPsiFactory(element)
+
+        val newPrimaryConstructor = factory.createDeclarationByPattern<KtClass>(
+                "class A constructor(\n $0, $1, $2\n)",
+                factory.createParameter("context: android.content.Context"),
+                factory.createParameter("attrs: android.util.AttributeSet? = null"),
+                factory.createParameter("defStyleAttr: Int = 0")
+        ).primaryConstructor!!
+
+        val primaryConstructor = ktClass.createPrimaryConstructorIfAbsent().replaced(newPrimaryConstructor)
+        ShortenReferences.DEFAULT.process(primaryConstructor.valueParameterList!!)
+
+        primaryConstructor.addAnnotation(fqNameAnnotation, whiteSpaceText = " ")
+
+        element.replace(factory.createSuperTypeCallEntry(element.text + "(context, attrs, defStyleAttr)"))
+    }
+
+    companion object Factory : KotlinSingleIntentionActionFactory() {
+
+        private val fqNameAnnotation = FqName("kotlin.jvm.JvmOverloads")
+
+        private val requireConstructorParameterTypes = listOf(
+                listOf("android.content.Context"),
+                listOf("android.content.Context", "android.util.AttributeSet"),
+                listOf("android.content.Context", "android.util.AttributeSet", "int")
+        )
+
+        override fun createAction(diagnostic: Diagnostic): IntentionAction? {
+            val superTypeEntry = SUPERTYPE_NOT_INITIALIZED.cast(diagnostic).psiElement
+
+            val ktClass = superTypeEntry.containingClass() ?: return null
+            if (ktClass.primaryConstructor != null) return null
+
+            val context = superTypeEntry.analyze()
+            val type = superTypeEntry.typeReference?.let { context[BindingContext.TYPE, it] } ?: return null
+
+            if (!type.isAndroidView() && type.supertypes().none { it.isAndroidView() }) return null
+
+            var declaration = type.getDeclaration()
+
+            if (declaration is KtClass) {
+                declaration = declaration.toLightClass()
```

Comment:
I think it could be implemented without light classes, they are pretty heavy and should be avoided if possible 
Try to get everything you need from descriptor
```Kotlin
val classDescriptor = type.constructor.declarationDescriptor as? ClassDescriptor ?: return null
val constructorParameters = classDescriptor.constructors.forEach {
    it.valueParameters... // some checking stuff
}
```